src/v0/destinations/klaviyo/transform.js

```diff
@@ -43,61 +56,81 @@ const { JSON_MIME_TYPE } = require('../../util/constant');
-  const response = defaultRequestConfig();
-  let personId;
-  if (message.channel !== 'sources') {
-    personId = await isProfileExist(message, destination);
-  }
+  let profileId;
   let propertyPayload = constructPayload(message, MAPPING_CONFIG[category.name]);
   // Extract other K-V property from traits about user custom properties
-  propertyPayload = extractCustomFields(
+  let customPropertyPayload = {};
+  customPropertyPayload = extractCustomFields(
     message,
-    propertyPayload,
+    customPropertyPayload,
     ['traits', 'context.traits'],
     WhiteListedTraits,
   );
-  if (!personId) {
-    propertyPayload = removeUndefinedAndNullValues(propertyPayload);
-    if (destination.Config?.enforceEmailAsPrimary) {
-      delete propertyPayload.$id;
-      propertyPayload._id = getFieldValueFromMessage(message, 'userId');
-    }
-    const payload = {
-      token: destination.Config.publicApiKey,
-      properties: propertyPayload,
-    };
-    response.endpoint = `${BASE_ENDPOINT}${category.apiUrl}`;
-    response.method = defaultPostRequestConfig.requestMethod;
-    response.headers = {
-      'Content-Type': JSON_MIME_TYPE,
-      Accept: 'text/html',
-    };
-    response.body.JSON = removeUndefinedAndNullValues(payload);
+
+  propertyPayload = removeUndefinedAndNullValues(propertyPayload);
+  if (enforceEmailAsPrimary) {
+    delete propertyPayload.external_id;
+    propertyPayload.properties._id = getFieldValueFromMessage(message, 'userId');
+  }
+  const data = {
+    type: 'profile',
+    attributes: propertyPayload,
+    properties: removeUndefinedAndNullValues(customPropertyPayload),
+  };
+  const payload = {
+    data: removeUndefinedAndNullValues(data),
+  };
+  const endpoint = `${BASE_ENDPOINT}${category.apiUrl}`;
+  const requestOptions = {
+    headers: {
+      Authorization: `Klaviyo-API-Key ${privateApiKey}`,
+      accept: JSON_MIME_TYPE,
+      'content-type': JSON_MIME_TYPE,
+      revision: '2023-02-22',
+    },
+  };
+  const resp = await httpPOST(endpoint, payload, requestOptions);
+  if (resp.response?.status === 201) {
+    profileId = resp.response?.data?.data?.id;
+  } else if (resp.response?.response?.status === 409) {
+    const { response } = resp.response;
+    profileId = response.data?.errors[0]?.meta?.duplicate_profile_id;
   } else {
-    response.endpoint = `${BASE_ENDPOINT}/api/v1/person/${personId}`;
-    response.method = defaultPutRequestConfig.requestMethod;
-    response.headers = {
-      Accept: JSON_MIME_TYPE,
-    };
-    response.params = removeUndefinedAndNullValues(propertyPayload);
-    response.params.api_key = destination.Config.privateApiKey;
+    throw new NetworkError(
+      `Failed to create user due to ${resp.response?.data}`,
+      resp.response?.response?.status,
+      {
+        [tags.TAG_NAMES.ERROR_TYPE]: getDynamicErrorType(resp.response?.response?.status),
+      },
+      resp.response?.data,
+    );
+  }
+
```

Comment:
Use handleHttpRequest to send request and handle response and also move this to aa utility func